test/e2e/workload_test.go

```diff
@@ -884,3 +886,325 @@ var _ = Describe("DaemonSet", func() {
+
+			By("Check for succeeded BackupSession")
+			f.EventuallyBackupSessionPhase(bs.ObjectMeta).Should(Equal(v1beta1.BackupSessionSucceeded))
+
+			By("Check for repository status updated")
+			f.EventuallyRepository(&rc).Should(WithTransform(f.BackupCountInRepositoriesStatus, BeNumerically(">=", 1)))
+
+			By("Delete BackupConfiguration")
+			err = f.DeleteBackupConfiguration(backupCfg)
+			err = framework.WaitUntilBackupConfigurationDeleted(f.StashClient, backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting to remove sidecar")
+			f.EventuallyReplicationController(rc.ObjectMeta).ShouldNot(matcher.HaveSidecar(util.StashContainer))
+			err = util.WaitUntilRCReady(f.KubeClient, rc.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+		}
+	)
+	Context("Backup && Restore for ReplicationController", func() {
+		BeforeEach(func() {
+			pvc = f.GetPersistentVolumeClaim()
+			err = f.CreatePersistentVolumeClaim(pvc)
+			Expect(err).NotTo(HaveOccurred())
+			rc = f.ReplicationController(pvc.Name)
+
+			targetref = v1beta1.TargetRef{
+				APIVersion: "v1",
+				Kind:       apis.KindReplicationController,
+				Name:       rc.Name,
+			}
+		})
+		AfterEach(func() {
+			err := f.DeleteReplicationController(rc.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			err = framework.WaitUntilReplicationControllerDeleted(f.KubeClient, rc.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = f.DeleteRepository(repo)
+			Expect(err).NotTo(HaveOccurred())
+			err = framework.WaitUntilRepositoryDeleted(f.StashClient, repo)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = f.DeleteRestoreSession(restoreSession.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+			err = framework.WaitUntilRestoreSessionDeleted(f.StashClient, restoreSession.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("General Backup new ReplicationController", func() {
+			By("Creating New ReplicationController Backup")
+			testRCBackup()
+
+			By("Remove sample data from workload")
+			err = f.CleanupSampleDataFromWorkload(rc.ObjectMeta, apis.KindReplicationController)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating Restore Session")
+			err = f.CreateRestoreSession(restoreSession)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for initContainer")
+			f.EventuallyReplicationController(rc.ObjectMeta).Should(matcher.HaveInitContainer(util.StashInitContainer))
+
+			By("Waiting for restore to succeed")
+			f.EventuallyRestoreSessionPhase(restoreSession.ObjectMeta).Should(Equal(v1beta1.RestoreSessionSucceeded))
+			err = util.WaitUntilRCReady(f.KubeClient, rc.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking the workload data has been restored")
+			restoredData, err = f.ReadSampleDataFromMountedDirectory(rc.ObjectMeta, framework.GetPathsFromRestoreSession(&restoreSession), apis.KindReplicationController)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(restoredData).To(Not(BeEmpty()))
```

Comment:
Use `Expect(restoredData).NotTo(BeEmpty())` instead of `Expect(restoredData).To(Not(BeEmpty()))`